UDPBroadcast/UDPv6BroadcastConnection.swift

```diff
@@ -0,0 +1,131 @@
+    let interface: String
+    
+    // MARK: Initializers
+    
+    /// Initializes the UDP connection with the correct port address.
+    
+    /// - Note: This doesn't open a socket! The socket is opened transparently as needed when sending broadcast messages. If you want to open a socket
+    ///   immediately, use the `bindIt` parameter. This will also try to reopen the socket if it gets closed.
+    ///
+    /// - Parameters:
+    ///   - interface: Name of the network interface, usually "en0"
+    ///   - port: Number of the UDP port to use.
+    ///   - bindIt: Opens a port immediately if true, on demand if false. Default is false.
+    ///   - handler: Handler that gets called when data is received.
+    ///   - errorHandler: Handler that gets called when an error occurs.
+    ///
+    /// - Throws: Throws a `ConnectionError` if an error occurs.
+    public init(interface: String = "en0", port: UInt16, bindIt: Bool = false, handler: ReceiveHandler?, errorHandler: ErrorHandler?) throws {
+
+        var addr = in6_addr()
+        let ret = withUnsafeMutablePointer(to: &addr) {
+            inet_pton(AF_INET6, UDPv6BroadcastConnection.LINK_LOCAL_NODES, UnsafeMutablePointer($0))
+        }
+
+        if ret == 0 {
+            throw ConnectionError.createv6AddressFailed(message: "Invalid address")
+        } else if ret == -1 {
+            var msg: String = ""
+            if let errorString = String(validatingUTF8: strerror(errno)) {
+                msg = errorString
+            }
+            throw ConnectionError.createv6AddressFailed(message: "Failed: \(msg) (\(errno))")
+        }
+        // addr contains the result.
+        
+        self.v6address = sockaddr_in6(
+            sin6_len: UInt8(MemoryLayout<sockaddr_in6>.size),
+            sin6_family: sa_family_t(AF_INET6),
+            sin6_port: UDPv6BroadcastConnection.htonsPort(port: port),
+            sin6_flowinfo: 0,
+            sin6_addr: addr,
+            sin6_scope_id: 0
+        )
+        self.interface = interface
+    
+        try super.init(bindIt: bindIt, handler: handler, errorHandler: errorHandler)
+    }
+    
+    override func setupSocket() throws -> Int32 {
+        
+        let newSocket = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP)
+        guard newSocket > 0 else { throw ConnectionError.createSocketFailed }
+        
+        // Enable broadcast on socket
+        let index = if_nametoindex(interface.cString(using: .ascii))
+        var scope: UInt32 = UInt32(index)
+        let ret = setsockopt(newSocket, IPPROTO_IPV6, IPV6_MULTICAST_IF, &scope, socklen_t(MemoryLayout<UInt32>.size));
+        if ret == -1 {
+            debugPrint("Couldn't enable broadcast on socket")
+            close(newSocket)
+            throw ConnectionError.enableBroadcastFailed
+        }
+
+        // Bind socket if needed
+        if shouldBeBound {
+            // FIXME: not yet implemented
+            debugPrint("FIXME: implement IPv6 binding!")
```

Comment:
Will this be done?

Reply:
Probably not. This was added to the original project by a 3rd party but we will never use it. I'm tempted to remove this addition to keep the whole thing as simple as possible.